lib/gemstash/cli/preload.rb

```diff
@@ -0,0 +1,31 @@
+require "gemstash"
+
+module Gemstash
+  class CLI
+    # This implements the command line preload task to cache all the available gems:
+    # $ gemstash preload
+    class Preload
+      include Gemstash::Env::Helper
+
+      def initialize(cli)
+        Gemstash::Env.current = Gemstash::Env.new
+        @cli = cli
+      end
+
+      def run
+        return unless are_you_sure?
+        http_client = HTTPClient.for(Upstream.new(@cli.options[:server_url]))
+        preloader = Gemstash::Preload::GemPreloader.new(http_client, latest: @cli.options[:latest]).
+          limit(@cli.options[:limit]).skip(@cli.options[:skip]).threads(@cli.options[:threads])
```

Comment:
ah, no, I am actually using the client to say and ask things. I need it for that.